### PR TITLE
Add prow config for ironic-rhcos-downloader

### PR DESCRIPTION
**Pull request by elfosardo**
Adding configuration for prow plus basic CI for the new forked
repo ironic-rhcos-downloader

### CODE REVIEW

File: ci-operator/config/openshift/ironic-rhcos-downloader/openshift-ironic-rhcos-downloader-release-4.2.yaml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  from: base-os
+  to: ironic-rhcos-downloader
+promotion:
+  disabled: true
+  name: "4.2"
+  namespace: ocp
+raw_steps:
+- pipeline_image_cache_step:
+    commands: |
+      curl http://base-openstack-4-2.ocp.svc > /etc/yum.repos.d/base-openstack-4-2.repo
```
Comment (russellb) [inline ci-operator/config/openshift/ironic-rhcos-downloader/openshift-ironic-rhcos-downloader-release-4.2.yaml]: What do we need from openstack for this image?  It looks like we just install `qemu-img` and `jq` from rpms?

Comment (elfosardo) [reply ci-operator/config/openshift/ironic-rhcos-downloader/openshift-ironic-rhcos-downloader-release-4.2.yaml]: This was a wrong copy-paste, I'll remove it

**Pull request merged**